File: perl/f40.loop.pl
#!/usr/bin/perl -w
# spin loop construction with last to exiting
while (1) {
  print "Enter your name (quit to exit): "; # prompt user
  chomp($answer = <>);                      # git input and trim newline
  next if $answer =~ /^[\t\s]*$/;           # skip loop if not answer entered
  last if $answer eq "quit";                # exit loop if user wants to quit
  print "Hello $answer!\n";                 # output results as not exiting
}

File: php/f40.loop.php
#!/usr/bin/env php
<?php
// spin loop as always true
//   break used to exit loop
do {
    echo "Enter your name (quit to Exit): "; # output prompt
    $answer = rtrim(fgets(STDIN));           # get input, trim newline
    # skip loop if not answer entered
    if (preg_match("/^[\s\t]*$/", $answer)) continue; 
    if ($answer == "quit") break;            # exit loop if user quits                           
    echo "Hello $answer!\n";                 # output results as not exiting
} while (1);
?>
